### ISSUE TITLE
Erlang 19 compilation error

### ISSUE THREAD

**Issue by TBK145** [Feuerlabs/exometer_core#80]
On Erlang 19 I get this compilation error:
```
src/exometer_report.erl:none: error in parse transform 'lager_transform': {function_clause,
                                             [{lager_transform,
                                               '-walk_ast/2-fun-0-',
                                               [{typed_record_field,
                                                 {record_field,251,
                                                  {atom,251,reporter}},
                                                 {type,251,union,
                                                  [{type,251,module,[]},
                                                   {atom,251,'_'}]}}],
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,62}]},
                                              {lists,map,2,
                                               [{file,"lists.erl"},
                                                {line,1239}]},
                                              {lager_transform,walk_ast,2,
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,62}]},
                                              {compile,
                                               '-foldl_transform/2-anonymous-2-',
                                               2,
                                               [{file,"compile.erl"},
                                                {line,958}]},
                                              {compile,foldl_transform,2,
                                               [{file,"compile.erl"},
                                                {line,960}]},
                                              {compile,
                                               '-internal_comp/4-anonymous-1-',
                                               2,
                                               [{file,"compile.erl"},
                                                {line,315}]},
                                              {compile,fold_comp,3,
                                               [{file,"compile.erl"},
                                                {line,341}]},
                                              {compile,internal_comp,4,
                                               [{file,"compile.erl"},
                                                {line,325}]}]}
```
I can't seem to find out what causes it or how to fix it.

**Comment by uwiger** [comment]
It's likely due to the introduction of some new abstract forms in OTP 19.0.

See e.g. https://github.com/uwiger/parse_trans/issues/19

I fixed this in the latest version of `parse_trans`, but lager has its own transform.

**Comment by TBK145** [comment]
So this is actually a lager bug?

**Comment by uwiger** [comment]
Fixed in lager 3.2.1, as far as I can tell.

**Comment by TBK145** [comment]
It turned out I was using a dependency which was using an old version of exometer, this error does not exist in master. I'm sorry for the confusion.
